src/rf/apps/workers/status_updates.py

```diff
@@ -10,49 +10,234 @@
+        status_validate_error__isnull=True)
+
+    # If all images are valid, then mark layer validation as complete.
+    if len(layer_images) == len(valid_layer_images) and len(layer_images) > 0:
+        return mark_layer_status_end(image.layer_id, enums.STATUS_VALID)
+    """
+    return True
+
+
+def mark_image_invalid(s3_uuid, error_message):
+    try:
+        image = LayerImage.objects.get(s3_uuid=s3_uuid)
+    except LayerImage.DoesNotExist:
+        return False
+
+    image = update_layerimage_status_end(image, enums.STATUS_VALIDATE,
+                                         error_message)
+    image.save()
+
+    return update_layer_status_end(image.layer_id, enums.STATUS_VALIDATE,
+                                   ERROR_MESSAGE_LAYER_IMAGE_INVALID)
+
+
+def mark_image_transfer_start(s3_uuid):
+    try:
+        image = LayerImage.objects.get(s3_uuid=s3_uuid)
+    except LayerImage.DoesNotExist:
+        return False
+
+    image = update_layerimage_status_start(image, enums.STATUS_UPLOAD)
+    image.save()
+    return mark_layer_status_start(image.layer_id, enums.STATUS_UPLOAD)
+
+
+def mark_image_transfer_failure(s3_uuid, error_message):
+    try:
+        image = LayerImage.objects.get(s3_uuid=s3_uuid)
+    except LayerImage.DoesNotExist:
+        return False
+
+    image = update_layerimage_status_end(image,
+                                         enums.STATUS_UPLOAD,
+                                         error_message)
+    image.save()
+    return mark_layer_status_end(image.layer_id,
+                                 enums.STATUS_UPLOAD,
+                                 ERROR_MESSAGE_LAYER_IMAGE_TRANSFER)
+
+
 def get_layer_id_from_uuid(s3_uuid):
     try:
         image = LayerImage.objects.get(s3_uuid=s3_uuid)
         return image.layer_id
     except LayerImage.DoesNotExist:
         return None
+
+
+def update_layer_status_start(layer, status):
+    value = datetime.now()
+    if status == enums.STATUS_UPLOAD:
+        layer.status_upload_start = (value if layer.status_upload_start
+                                     is None else layer.status_upload_start)
+
+    elif status == enums.STATUS_VALIDATE:
+        layer.status_validate_start = (value if layer.status_validate_start
+                                       is None else layer.status_upload_start)
+
+    elif status == enums.STATUS_THUMBNAIL:
+        layer.status_thumbnail_start = (value if layer.status_thumbnail_start
+                                        is None else layer.status_upload_start)
+
+    elif status == enums.STATUS_CREATE_CLUSTER:
+        layer.status_create_cluster_start = (value if
+                                             layer.status_create_cluster_start
+                                             is None else
+                                             layer.status_upload_start)
+
+    elif status == enums.STATUS_CHUNK:
+        layer.status_chunk_start = (value if layer.status_upload_start
```

Comment:
Should `layer.status_upload_start` be replaced with `layer.status_chunk_start` here? Same thing for the next status.